source/features/hide-useless-comments.tsx

```diff
@@ -55,11 +57,112 @@ function unhide(event: React.MouseEvent<HTMLButtonElement>): void {
 	event.currentTarget.parentElement!.remove();
 }
 
+async function getMutedUsers(): Promise<string[]> {
+	return ((await optionsStorage.getAll() as Options).mutedUsers).split(' ');
+}
+
+async function setMutedUsers(mutedUsers: string[]): Promise<void> {
+	return optionsStorage.set({mutedUsers: mutedUsers.join(' ')});
+}
+
+function minimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.remove('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.add('d-none');
+}
+
+function unminimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
+
+async function onMuteUnmuteClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	let mutedUsers = await getMutedUsers();
+
+	const comment = (event.target as HTMLElement).closest('.js-comment-container')!;
+	const viewportOffset = comment.getBoundingClientRect().top;
+	const user = select('.author', comment)!.textContent!;
+
+	if (mutedUsers.includes(user)) {
+		mutedUsers = mutedUsers.filter(mutedUser => mutedUser !== user);
+	} else {
+		mutedUsers.push(user);
+	}
+
+	await setMutedUsers(mutedUsers);
```

Comment:
We don’t need to await this right here, it can be done later.

Reply:
Now we have to as we are immediately querying it again in `minimizeMutedUserComments` on the next line.